app/assets/javascripts/account/directives/myApplicationDirective.js.coffee

```diff
@@ -54,11 +60,51 @@ angular.module('dahlia.directives')
       scope.isSubmitted = ->
         ShortFormApplicationService.applicationWasSubmitted(scope.application)
 
+      scope.submittedWithLotteryResults = ->
+        scope.isSubmitted() && scope.listing.Lottery_Results
+
       scope.isPastDue = ->
         moment(scope.listing.Application_Due_Date) < moment()
 
       scope.lotteryNumber = ->
-        { lotteryNumber: scope.application.lotteryNumber }
+        if scope.listing.Lottery_Results
+          html = """
+            <button class='button-link lined' ng-click='viewLotteryResults()'>
```

Comment:
@callaghanc is there a reason why you have to return an html here? Can you just return the lottery number and do the ng-if on the markup for a button or to display the number?

Reply:
This is Dave's work. I don't have immediate insight into his reasoning here without doing further investigation which I am out of time to do this week. I believe the best way to understand this further would be to try the alternative you propose in your comment and see if that works.

Reply:
Hm then he must have tried it my way and found an obstacle to doing it that way - coz we always go for the standard way first. We have a few refactoring chores in the web app in the pipeline.. I'll include this in it so that we can merge.